src/models/data-state.model.ts

```diff
@@ -0,0 +1,2 @@
+type dataState = 'beforeLoad' | 'Loading' | 'Loaded' | 'Error';
```

Comment:
you can skip this file, and use Enum as a type

you can keep all Enums/Interfaces/Types from one section/module in just one file ex. data-state.model.ts